### PR TITLE
enable mypy for examples/workflow

### PR DESCRIPTION
**Pull request by sandalns**
### Linked issue(s):
KOL-4575
### What change does this PR introduce and why?
Enables the temporarily removed mypy checks from example/workflow and adds them back. mypy previously was also not ran on the `experimental/classification` or experimental/object_detection` directories.

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

**Comment by sandalns** [comment]
Main reasons for `type: ignores`

1) I ignored errors around numpy return and parameter types after time boxing 1hr and not making progress
2) Many errors around the expected type being `kolena.workflow.TestSample` and not `example.workflow.TestSample` etc

Any suggestions here would be appreciated.

**Review by grant-Kolena — APPROVED** [review]

File: kolena/_experimental/object_detection/workflow.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 try:
     from typing import Literal
 except ImportError:
-    from typing_extensions import Literal
+    from typing_extensions import Literal  # type: ignore
```
Comment (yifanwu-kolena) [inline kolena/_experimental/object_detection/workflow.py]: `typing.Literal` was introduced in python 3.8, considering we already dropped python 3.7 support after 1.0, i think we can remove these try catch everywhere

Comment (sandalns) [reply kolena/_experimental/object_detection/workflow.py]: Will remove

Comment (sandalns) [reply kolena/_experimental/object_detection/workflow.py]: done

File: examples/workflow/automatic_speech_recognition/automatic_speech_recognition/utils.py [hunk 2]
```diff
@@ -89,13 +89,13 @@ def generate_diff_word_level(reference: str, candidate: str) -> Dict[str, Any]:
...
 
-def preprocess_transcription(txt: Union[GroundTruth, Inference]) -> str:
+def preprocess_transcription(txt: str) -> str:
     """
     Preprocesses and standardizes text to prepare for metrics evaluations.
     Removes punctuation, changes text to lower case, and converts all NumWords to Numbers.
     """
     num = NumWordsToNum()
-    txt = re.sub(r"[^\w\s]", "", txt.transcription.label.lower())
+    txt = re.sub(r"[^\w\s]", "", txt)
```
Comment (yifanwu-kolena) [inline examples/workflow/automatic_speech_recognition/automatic_speech_recognition/utils.py:98]: based on the usage of `preprocess_transcription`(examples/workflow/automatic_speech_recognition/automatic_speech_recognition/evaluator.py), i don't think we should modify the code at here

```py
def compute_test_sample_metrics(gt: GroundTruth, inf: Inference) -> TestSampleMetric:
    gt = preprocess_transcription(gt)
    inf = preprocess_transcription(inf)
```

Comment (yifanwu-kolena) [reply examples/workflow/automatic_speech_recognition/automatic_speech_recognition/utils.py:98]: nvm, i noticed that we updated the usage part as well

**Review by yifanwu-kolena — APPROVED** [review]

**Pull request merged**